src/test/regress/expected/validate_constraint.out

```diff
@@ -0,0 +1,254 @@
+-- until it is validated by using the VALIDATE CONSTRAINT option.
+CREATE SCHEMA validate_constraint;
+SET search_path TO 'validate_constraint';
+SET citus.shard_replication_factor TO 1;
+SET citus.shard_count TO 8;
+SET citus.next_shard_id TO 8000000;
+SET citus.next_placement_id TO 8000000;
+CREATE TYPE constraint_validity AS (name text, validated bool);
+SELECT run_command_on_workers(
+         $$CREATE TYPE constraint_validity AS (name text, validated bool)$$);
+      run_command_on_workers       
+-----------------------------------
+ (localhost,57637,t,"CREATE TYPE")
+ (localhost,57638,t,"CREATE TYPE")
+(2 rows)
+
+CREATE VIEW constraint_validations_in_workers AS
+  SELECT (json_populate_record(NULL :: constraint_validity,
+                               json_array_elements_text((run_command_on_workers($$
+    SELECT
+      COALESCE(json_agg(row_to_json(d)), '[]'::json)
+    FROM
+      (
+        SELECT conname as name,
+        convalidated as validated
+        FROM pg_catalog.pg_constraint AS con
+        JOIN pg_catalog.pg_namespace AS ns
+        ON ns.oid = con.connamespace
+        WHERE nspname = 'validate_constraint'
+        AND contype = 'c'
+      )
+      d $$)).RESULT :: json) :: json)).*;
+CREATE VIEW constraint_validations AS
+  SELECT conname as "Constraint",
+         convalidated as "Validated?"
+  FROM pg_catalog.pg_constraint AS con
+         JOIN pg_catalog.pg_namespace AS ns ON ns.oid = con.connamespace
+  WHERE nspname = 'validate_constraint'
+    AND contype = 'c';
+CREATE TABLE referenced_table (
+  id          int UNIQUE,
+  test_column int
+);
+SELECT create_reference_table('referenced_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+CREATE TABLE constrained_table (
+  id                 int,
+  constrained_column int,
+  ref_id             int
+);
+SELECT create_distributed_table('constrained_table', 'ref_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- The two constraint types that are allowed to be NOT VALID
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT check_constraint CHECK (constrained_column > 100) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT check_constraint;
+ROLLBACK;
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT fk_constraint FOREIGN KEY (ref_id) REFERENCES referenced_table (id) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT fk_constraint;
+ROLLBACK;
+-- It is possible that some other table_constraint commands will support NOT VALID option in the future
+-- These should fail as Postgres does not yet allow these constraints to be NOT VALID
+-- If one of these queries are not failing on a future Postgres version, we will need to test corresponding VALIDATE CONSTRAINT queries
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT unique_constraint UNIQUE (constrained_column) NOT VALID;
+ERROR:  UNIQUE constraints cannot be marked NOT VALID
```

Comment:
I think we should remove the constraints which cannot be marked as NOT VALID, such as UNIQUE

Reply:
I think what disturbs me here is not having the commands that might be supported in the future. But, mostly trying to execute ` VALIDATE CONSTRAINT ` command after a failure in the transaction. Because, that only adds some verbose output to the test and hard to track. We could even remove `BEGIN/ROLLBACK` steps in the failing tests and only keep `ALTER TABLE .. ADD CONSTRAINT .. NOT VALID` commands.

Reply:
Valid point. I am removing the statements that are not executed in aborted transactions, and those `BEGIN/ROLLBACK` steps.

I guess I was a little bit confused about when it is appropriate to use those transaction blocks in a test. This task helped me get a better understanding. 

Correct me if I am wrong, but I think that it is better to use a `BEGIN/ROLLBACK` block when we wish to run a query that makes a successful change in the schema but we wish to revert it to test some other thing afterwards. (Which is not the case here, so removing them makes perfect sense)